src/util/initialState.ts

```diff
@@ -60,6 +61,7 @@ export interface State {
   expandHoverBottomPaths: Index<Path>,
   expandHoverTopPath?: Path | null,
   expandedContextThought?: Path,
+  focusNodePath?: number[] | null,
```

Comment:
> * Do these both represent the current caret position? How are they kept up-to-date when the caret moves?

Sometimes we want to set the selection at the specific place in the editable. So we need `focusNodePath` and `offset` to do that.

Reply:
> * Can `cursorFocusPosition` be derived from `focusNodePath`, or is it independent?

In many cases we just want to set the focus at the end or the start of the editable. That doesn't require `focusNodePath` and `offset`. So `cursorFocusPosition` is basically an enum with values `start` and `end`. If it is provided it will just override the  `focusNodePath` and `offset` and set the focus at the start or the end of the editable. Similar to `end` parameter in the previous implementation of `setSelection`.

Reply:
Okay, but why is this part of the global state? The way you described them they seem more like arguments to a reducer, i.e. a specific action instead of a persistent value.

Reply:
> Okay, but why is this part of the global state? The way you described them they seem more like arguments to a reducer, i.e. a specific action instead of a persistent value. 

You are correct we can just use ‘focusNodePath‘ and ‘offset‘ to keep selection at end or start too. I will remove this from.the global state. Thanks!